crates/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -436,6 +436,8 @@ where
                 } = result_value;
                 match value {
                     None => (),
+                    // disable this check for dev inspect
+                    Some(_) if Mode::allow_arbitrary_values() => (),
```

Comment:
nit: thoughts on wrapping the overall loop in `if !Mode::allow_arbitrary_values()` instead of skipping in each iteration of the inner loop?